### PR TITLE
NgModel support for inputs

### PR DESCRIPTION
**Pull request by Nexyll**
- [x]  Implement 
- [x]  Add demo in Angular
- [x]  ~~Tests~~ Components wrap native so tests are not our part.

### CODE REVIEW

**Comment by MonsieurMan** [comment]
Issue at #35

File: src/components/input/input.tsx [hunk 1]
```diff
@@ -119,23 +129,21 @@ export class Input {
...
     }
 
     @Watch('value')
     protected valueChanged() {
         this.wcsChange.emit({ value: this.value });
-        console.log(this.value);
     }
 
-    private onInput(ev: Event) {
+    private onInput = (ev: Event) => {
```
Comment (MonsieurMan) [inline src/components/input/input.tsx]: Declaring methods as fat arrow functions results in them not using the same memory space as normal ones and is heavier.

Is there any reason it has been changed ?

Also, `KeyboardEvent` should be used instead.

File: src/components/input/input.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    expect(changeSpy).toHaveReceivedEventTimes(1);
+    console.dir(changeSpy.firstEvent.detail);
+    expect(changeSpy).toHaveReceivedEventDetail({ value: 'A' });
+});
+
+xit(`Propagates wcsInput event when a key stroke occurs on the native input`, async () => {
+    // Given
+    const page = await newE2EPage({
+        html: `
+                <wcs-input />
```
Comment (MonsieurMan) [inline src/components/input/input.e2e.ts]: Custom elements are not intended to be used as self-closing, browser behavior is UB on this.

File: src/components/input/input.tsx (hunk 1 above)
Comment (Nexyll) [reply src/components/input/input.tsx]: Yes, the method must send an event from a property in the input component, so it had to remain in its scope.

Comment (MonsieurMan) [reply src/components/input/input.tsx]: https://stenciljs.com/docs/templating-jsx

The solution is to keep a normal function and provide the scope in the render function instead :

```jsx
onInput={(evt: KeyboardEvent) => this.onInput(evt)}
// or
onInput={this.onInput.bind(this)}
```

**Pull request merged**